### PR TITLE
Added toolbar with status and drafting workflow

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| License | MIT
| Fixed | #8 

#### What's in this PR?

Added whole toolbar functions (unpublish, status, ..)

### CODE REVIEW

File: Resources/public/js/components/articles/edit/seo/main.js [hunk 1]
```diff
@@ -7,7 +7,9 @@
  * with this source code in the file LICENSE.
  */
 
-define(function() {
+define([
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/seo/main.js]: should fit into a single line

File: Resources/public/js/components/articles/list/main.js [hunk 2]
```diff
@@ -165,7 +170,29 @@ define(['underscore'], function(_) {
...
                     viewOptions: {
                         table: {
-                            actionIconColumn: 'title'
+                            actionIconColumn: 'title',
+                            badges: [
+                                {
+                                    column: 'title',
+                                    callback: function(item, badge) {
+                                        var icons = '',
+                                            tooltip = this.sandbox.translate(this.translations.unpublished);
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/list/main.js]: this.translations is already translated you can omit the sandbox here.

**Comment by trickreich** [comment]
@wachterjohannes wished changes are implemented

File: Resources/public/js/components/articles/edit/main.js [hunk 3]
```diff
@@ -292,7 +292,7 @@ define([
                 this.sandbox.emit(
                     'sulu.header.tabs.label.show',
                     this.sandbox.util.sprintf(
-                        this.sandbox.translate(this.defaults.translations.draftLabel),
+                        this.sandbox.translate(this.translations.draftLabel),
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: you can omit the sandbox here (:

File: Resources/public/js/components/articles/edit/main.js [hunk 4]
```diff
@@ -301,7 +301,7 @@ define([
                     [
                         {
                             id: 'delete-draft',
-                            title: this.sandbox.translate(this.defaults.translations.removeDraft),
+                            title: this.sandbox.translate(this.translations.removeDraft),
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: omit sandbox

File: Resources/public/js/components/articles/edit/main.js [hunk 5]
```diff
@@ -222,11 +297,56 @@ define([
...
+                    }.bind(this)).fail(function() {
+                        this.sandbox.emit('husky.label.header.reset');
+                        this.sandbox.emit(
+                            'sulu.labels.error.show',
+                            'labels.error.remove-draft-desc',
+                            'labels.error'
+                        );
+                    }.bind(this));
+                }.bind(this),
+                this.sandbox.translate(this.defaults.translations.deleteDraftConfirmTitle),
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: dont use it like this (: in the whole PR! `this. translations.deleteDraftConfirmTitle`

File: Resources/public/js/components/articles/edit/main.js [hunk 6]
```diff
@@ -61,12 +85,44 @@ define([
                         }
                     }
                 }
-            };
+            }
+
+            if (SecurityChecker.hasPermission(this.data, 'live')) {
+                editDropdown.unpublish = {
+                    options: {
+                        title: this.sandbox.translate('sulu-document-manager.unpublish'),
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: use `defaults.translations`.

File: Resources/public/js/components/articles/edit/main.js [hunk 7]
```diff
@@ -222,11 +297,56 @@ define([
                             changed: this.sandbox.date.format(this.data.changed, true),
                             user: response.username
                         }
-                    )
+                    ),
+                    [
+                        {
+                            id: 'delete-draft',
+                            title: this.sandbox.translate(this.translations.removeDraft),
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: omit sandbox here

**Comment by trickreich** [comment]
@wachterjohannes should be ok now

File: Resources/public/js/components/articles/edit/main.js [hunk 8]
```diff
@@ -268,6 +389,65 @@ define([
...
+                        this.saved(response.id, response);
+                    }.bind(this)).fail(function() {
+                        this.sandbox.emit(
+                            'sulu.labels.error.show',
+                            'labels.error.content-unpublish-desc',
+                            'labels.error'
+                        );
+                    }.bind(this));
+                }.bind(this),
+                title: this.defaults.translations.unpublishConfirmTitle,
```
Comment (wachterjohannes) [inline Resources/public/js/components/articles/edit/main.js]: same for two others in this file

**Comment by wachterjohannes** [comment]
One small issue i have found. When i try to go to a "nicht mehr veröffentlichen" page on the website i get following error 
image

I think we have to check `$object instanceof ArticleDocument`

**Pull request merged**